### PR TITLE
Change coordinate field

### PR DESCRIPTION
**Pull request by xacobofg**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

I have a plugin that performs calculations using the coordinates of the locations but as the coordinates are saved as varchar it generates warnings in the logs.
I also checked that it is recommended that the coordinate fields should be saved as decimal.

### CODE REVIEW

**Comment by keguira** [comment]
Quick question : why change the default value from "empty" with NULL (impliying "undefined") to 0.0 which is a real existing coordinate and sometimes used as an "error" value or used by people as default coordinates to not share it ?

File: install/migrations/update_10.0.7_to_10.0.8/entity.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * @var DB $DB
+ * @var Migration $migration
+ */
+$migration->changeField(
+    Entity::getTable(),
+    'latitude',
+    'latitude',
+    "decimal(9,6) NOT NULL DEFAULT '0.0000'",
```
Comment (keguira) [inline install/migrations/update_10.0.7_to_10.0.8/entity.php:44]: Quick question : why change the default value from "empty" with NULL (impliying "undefined") to 0.0 which are real existing coordinates or altitude and sometimes used as an "error" value or used by people as default coordinates to not share it ?

I already faced such behavior in other applications. 0 is not empty such as null is neither true or false.

File: install/mysql/glpi-empty.sql [hunk 2]
```diff
@@ -2785,9 +2785,9 @@ CREATE TABLE `glpi_entities` (
   `contracts_id_default` int unsigned NOT NULL DEFAULT '0',
   `enable_custom_css` int NOT NULL DEFAULT '-2',
   `custom_css_code` text,
-  `latitude` varchar(255) DEFAULT NULL,
-  `longitude` varchar(255) DEFAULT NULL,
-  `altitude` varchar(255) DEFAULT NULL,
+  `latitude` decimal(9,6) NOT NULL DEFAULT '0.0000',
+  `longitude` decimal(9,6) NOT NULL DEFAULT '0.0000',
+  `altitude` decimal(9,6) NOT NULL DEFAULT '0.0000',
```
Comment (cedric-anne) [inline install/mysql/glpi-empty.sql:2790]: Why did you added the `NOT NULL` option? It is probably preferable to keep a null value when coordinates are not defined.

```suggestion
  `latitude` decimal(9,6) DEFAULT NULL,
  `longitude` decimal(9,6) DEFAULT NULL,
  `altitude` decimal(9,6) DEFAULT NULL,
```

Comment (xacobofg) [reply install/mysql/glpi-empty.sql]: Sorry for the delay. I changed the fields to be NULL by default.

**Comment by keguira** [comment]
> Also I don't remember how many decimals open street map handles and if the definition you use (9,6) is correct for backward compatibility

After some digging : 
* [openstreetmap node doc](https://wiki.openstreetmap.org/wiki/Node) says it handles 7 digits for the fractional part. 
* after tests : the "define location here" button on a map can return much more digits. By zooming on the map i can define a location with lat/long with 14 digits for the fractional part.
* users may have recorded location by reporting data from googlemap for example which can also use more than 6 digits for the fractional part

I don't know the rule of precision we want to apply but today, there is none in GLPI such as user can store very "accurate" location.

By changing the type for `Decimal(9,6)`, we will round some data in a such way we will change a behavior and maybe break some user data.
In this PR, we should change the manual input according to the selected rule of precision to disallow contribution of decimal with to much precision. We should also change how it is contributed with the "change location to here" button.

It should not be a "fix" but an enhancement for next 10.1 version.

File: install/migrations/update_10.0.7_to_10.0.8/entity.php [hunk 3]
```diff
@@ -41,17 +41,17 @@
     Entity::getTable(),
     'latitude',
     'latitude',
-    "decimal(9,6) NOT NULL DEFAULT '0.0000'",
+    "decimal(9,6) DEFAULT NULL",
```
Comment (keguira) [inline install/migrations/update_10.0.7_to_10.0.8/entity.php:44]: > Also I don't remember how many decimals open street map handles and if the definition you use (9,6) is correct for backward compatibility

After some digging : 
* [openstreetmap node doc](https://wiki.openstreetmap.org/wiki/Node) says it handles 7 digits for the fractional part. 
* after tests : the "define location here" button on a map can return much more digits. By zooming on the map i can define a location with lat/long with 14 digits for the fractional part.
* users may have recorded location by reporting data from googlemap for example which can also use more than 6 digits for the fractional part

I don't know the rule of precision we want to apply but today, there is none in GLPI such as user can store very "accurate" location.

By changing the type for `Decimal(9,6)`, we will round some data in a such way we will change a behavior and maybe break some user data.
In this PR, we should change the manual input according to the selected rule of precision to disallow contribution of decimal with to much precision. We should also change how it is contributed with the "change location to here" button.

It should not be a "fix" but an enhancement for next 10.1 version.

Comment (trasher) [reply install/migrations/update_10.0.7_to_10.0.8/entity.php:44]: Yes, required changes should target main branch for 10.1 release.

**Review by trasher — CHANGES_REQUESTED** [review]
See @keguira comments; precision seems incorrect, and this change would also require code changes.

**Comment by trasher** [comment]
Current PR is stalled from a while, I'll close it.

Feel free to reopen if you're going to work again on that subject. Thank you.

**Pull request closed without merging**